### PR TITLE
Calculate `localize`d path for remote file

### PR DESCRIPTION
**Pull request by annasong20**
This PR calculates the localized, or locally saved, path (relative to the current directory in the localize destination) for a remote file in `kustomize localize`.

### CODE REVIEW

File: api/internal/localizer/util.go [hunk 1]
```diff
@@ -121,7 +122,20 @@ func cleanFilePath(fSys filesys.FileSystem, root filesys.ConfirmedDir, file stri
...
+	if err != nil {
+		log.Fatalf("cannot parse validated file url %q: %s", fileURL, err.Error())
+	}
+
+	// preserve percent-encodings in case of sub-delims special meaning
+	// remove extraneous '..' parent directory dot-segments
+	path := filepath.Join(string(filepath.Separator), filepath.FromSlash(u.EscapedPath()))
+
+	// raw github urls are the only type of file urls kustomize officially accepts, in which case
+	// path consists of org, repo, version, path in repo
```
Comment (yuwenma) [inline api/internal/localizer/util.go:137]: qq: Which line of code does this comment applies to? Could it be more readable to summarize each comments and give a method level docstring?
e.g. 
```
// locFilePath converts a URL to localized form. e.g. XXXX -> YYYYY
func locFilePath()
```

File: api/internal/localizer/util_test.go [hunk 2]
```diff
@@ -16,3 +18,49 @@ func TestUrlBase(t *testing.T) {
 func TestUrlBaseTrailingSlash(t *testing.T) {
 	require.Equal(t, "repo", urlBase("github.com/org/repo//"))
 }
+
+// simpleJoin is filepath.Join() without side effects
```
Comment (yuwenma) [inline api/internal/localizer/util_test.go:22]: nit: Can you add a link or explain the "side effects" this function solves?

**Review by yuwenma — APPROVED** [review]
Nice PR. Two minor comments

File: api/internal/localizer/util.go (hunk 1 above)
Comment (annasong20) [reply api/internal/localizer/util.go:137]: I
* moved that comment
* added the summary and example to the spec

File: api/internal/localizer/util.go [hunk 3]
```diff
@@ -120,8 +121,25 @@ func cleanFilePath(fSys filesys.FileSystem, root filesys.ConfirmedDir, file stri
...
+// locFilePath converts a URL to its localized form, e.g.
+// https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/api/krusty/testdata/localize/simple/service.yaml ->
+// localized-files/raw.githubusercontent.com/kubernetes-sigs/kustomize/master/api/krusty/testdata/localize/simple/service.yaml.
+//
+// fileURL must be a validated file URL.
+func locFilePath(fileURL string) string {
+	// File urls must have http or https scheme, so it is safe to use url.Parse.
+	u, err := url.Parse(fileURL)
+	if err != nil {
+		log.Fatalf("cannot parse validated file url %q: %s", fileURL, err.Error())
```
[the remark this answers is not in the document]
Comment (annasong20) [reply api/internal/localizer/util.go:133]: Yes. `url.Parse` is called on all remote file urls in `fileLoader`'s `Load` method.

**Pull request merged**